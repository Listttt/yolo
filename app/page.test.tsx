/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import Page from "./page";
import StoreProvider from "./StoreProvider";

describe('Main page', () => {
    it.only("renders homepage unchanged", () => {
        const { container } = render(
            <StoreProvider>
                <Page/>
            </StoreProvider>
        );

        expect(container).toMatchSnapshot();
    });

    it.only('contains DataTable', () => {
        render(
            <StoreProvider>
                <Page/>
            </StoreProvider>
        )
        expect(screen.getByTestId('data-table')).toBeInTheDocument();
    });

    // Can't win AsyncThunk dispatch testing
    it.skip('dispatches "fetchCountries" action', () => {
    });

    it.todo('gets countries from store');
});
