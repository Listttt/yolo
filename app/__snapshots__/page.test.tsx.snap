// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page renders homepage unchanged 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <div
      class="rounded-lg border text-card-foreground shadow-sm w-[350px] p-8 bg-secondary"
    >
      <div
        data-testid="data-table"
      >
        <div
          class="flex items-center py-4"
        >
          <input
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 max-w-sm"
            placeholder="Filter by code"
            value=""
          />
        </div>
        <div
          class="rounded-md border"
        >
          <div
            class="relative w-full overflow-auto"
          >
            <table
              class="w-full caption-bottom text-sm"
            >
              <thead
                class="[&_tr]:border-b"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    code
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    country
                  </th>
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                    colspan="2"
                  >
                    No results.
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
